sbin/spark-daemon.sh

```diff
@@ -162,6 +162,15 @@ run_command() {
   esac
 
   echo "$newpid" > "$pid"
+
+  for i in {1..10}
+  do
+    if [[ $(ps -p "$newpid" -o comm=) =~ "java" ]]; then
+       break
+    fi
+    sleep 0.5
```

Comment:
Maybe once a second for 5 seconds? you need to remove the "sleep 2" below.
Maybe you can use a variable rather than re-test the condition after the loop?

However, the PR has also been messed up with some unrelated changes. Can you fix or else open a new PR?